mso/resource_mso_schema_template_contract.go

```diff
@@ -524,7 +524,10 @@ func resourceMSOTemplateContractRead(d *schema.ResourceData, m interface{}) erro
 	if err != nil {
 		return errorForObjectNotFound(err, d.Id(), schemaCont, d)
 	}
-	setContractFromSchema(d, schemaCont, schemaId, templateName, contractName)
+	err = setContractFromSchema(d, schemaCont, schemaId, templateName, contractName)
+	if err != nil {
+		d.SetId("")
```

Comment:
It may also be useful to log any errors from the `setContractFromSchema` function.
If they are expected errors maybe just log them as debug messages.

Reply:
```suggestion
                log.Printf("[DEBUG] %v", err)
		d.SetId("")
```

Reply:
fixed